### PR TITLE
[flutter_tools] remove mock from analyze_size_test.dart

### PR DESCRIPTION
**Pull request by jonahwilliams**
As it says

Work towards https://github.com/flutter/flutter/issues/71511

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/base/analyze_size_test.dart [hunk 1]
```diff
@@ -224,4 +224,7 @@ void main() {
   });
 }
 
-class MockUsage extends Mock implements Usage {}
+class FakeUsage extends Fake implements Usage {
+  @override
+  void sendEvent(String category, String parameter, {String label, int value, Map<String, String> parameters}) { }
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/base/analyze_size_test.dart:229]: We don't even have any expectations for it, just need it to not throw

**Review by jmagman — COMMENTED** [review]
Is there something wrong with `Usage.test()` in this case?  You know, for when we need to migrate off the `test/fake` dependency for some annoying reason.

**Comment by jonahwilliams** [comment]
its prints to real stdout which is obnoxious :P 

Could we make the test version store the events in a big List<Object>, like we do for buffer logger?

**Comment by jmagman** [comment]
> its prints to real stdout which is obnoxious :P
> 
> Could we make the test version store the events in a big List, like we do for buffer logger?

Oh yeah, there is something wrong.  The problem is we have no control over how that usage code works.
https://github.com/dart-lang/usage/blob/6c64d9e7b6b3758d06d030efcb5afe20bfc04dde/lib/usage.dart#L233-L237

I would love to get something better than catching the logs in a zone.

https://github.com/flutter/flutter/blob/bf37a3964b800708f3efd823b2ee791acf4ef190/packages/flutter_tools/test/commands.shard/hermetic/build_ipa_test.dart#L244-L252
https://github.com/flutter/flutter/blob/d1538320d21c67ac5f90ae1b1b4ca6c76307aebe/packages/flutter_tools/test/src/common.dart#L98-L107

I tried to work around it for awhile but I couldn't come up with a way to inject something into the usage package to capture it.

**Comment by jonahwilliams** [comment]
We should make our own analytics.test version then. I think I have a better idea

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**